File: Irrgarten/src/irrgarten/Weapon.java
package irrgarten;

/**
 * Represents a weapon from the game Irrgarten

 */
/**
 * The Weapon class represents a weapon with a certain power and number of uses.
 */
public class Weapon extends CombatElement {
    /**
     * Parameterized constructor to create a Weapon object.
     * 
     * @param power the power of the weapon
     * @param uses the number of uses left for the weapon
     */
    public Weapon(float power, int uses) {
        super(power,uses);
    }
    
    /**
     * Makes the weapon attack. If the weapon has no uses left, the power is 0.
     * 
     * @return the power of the weapon as a float point number
     */
    public float attack() {
        return this.produceEffect();
    }
    
    /**
     * Returns a string representation of the weapon.
     * 
     * @return a string with information about the weapon in the format "W[power, uses]"
     */
    @Override
    public String toString() {
        return "W" + super.toString();
    }
}

File: Irrgarten_ruby/Weapon.rb
#encoding:utf-8
module Irrgarten

class Weapon
    
    # Parameterized constructor
    # power power of the weapon
    # uses left for the weapon
    def initialize (power, uses)
        # Here the instance fields are being initialized
        @power = power;
        @uses = uses;
    end

    # Returns a float number representing the power of the weapon
    def attack 
        if (@uses > 0)
            @uses=@uses-1
            return @power
        else
            return 0
        end
    end
    
    # Decides if a weapon must be descarted, using the method DiscardElement
    # of the Dice class
    def discard 
        return Dice.discard_element(@uses)
    end 

    # Print the uses and power of the weapon
    # Return a string containing the information of the weapon
    def to_s
        return "W["+@power.to_s+","+@uses.to_s+"]"
    end

end

end
